core/src/test/scala/unit/kafka/coordinator/transaction/TransactionStateManagerTest.scala

```diff
@@ -337,6 +337,62 @@ class TransactionStateManagerTest {
     verifyWritesTxnMarkersInPrepareState(PrepareAbort)
   }
 
+  @Test
+  def shouldRemoveExpiredTransactions(): Unit = {
```

Comment:
Related to the above comment, let's test for the cases that after we have decided to expire a txnId, another `addPartition` is received and handled.